### PR TITLE
port process-status, set-process-buffer, process-send-string

### PR DESCRIPTION
**Pull request by brotzeit**

### CODE REVIEW

File: rust_src/src/process.rs [hunk 1]
```diff
@@ -102,6 +116,102 @@ pub fn set_process_plist(process: LispObject, plist: LispObject) -> LispObject {
...
+/// listen -- for a network stream server that is listening.
+/// closed -- for a network stream connection that is closed.
+/// connect -- when waiting for a non-blocking connection to complete.
+/// failed -- when a non-blocking connection has failed.
+/// nil -- if arg is a process name and no such process exists.
+/// PROCESS may be a process, a buffer, the name of a process, or
+/// nil, indicating the current buffer's process.
+#[lisp_fn]
+pub fn process_status(process: LispObject) -> LispObject {
+    let mut p = process;
```
Comment (shaleh) [inline rust_src/src/process.rs]: Could this be:

    let p = if process.is_string() {
        get_process(process)
    } else {
       LispObject::from( unsafe {  cget_process(process.to_raw()) } )
    }

File: rust_src/src/process.rs [hunk 2]
```diff
@@ -102,6 +116,102 @@ pub fn set_process_plist(process: LispObject, plist: LispObject) -> LispObject {
...
+/// Return BUFFER.
+#[lisp_fn]
+fn set_process_buffer(process: LispObject, buffer: LispObject) -> LispObject {
+    let mut p_ref = process.as_process_or_error();
+    if buffer.is_not_nil() {
+        buffer.as_buffer_or_error();
+    }
+    p_ref.set_buffer(buffer);
+    let process_type = LispObject::from(p_ref.process_type);
+    if process_type.eq(LispObject::from(Qnetwork)) || process_type.eq(LispObject::from(Qserial))
```
Comment (shaleh) [inline rust_src/src/process.rs]: Why drop the named predicates?

File: rust_src/src/process.rs [hunk 3]
```diff
@@ -39,6 +52,7 @@ fn get_process(name: LispObject) -> LispObject {
         name
     } else {
         name.as_string_or_error();
+        ();
```
Comment (Wilfred) [inline rust_src/src/process.rs]: What's this for?

File: rust_src/src/process.rs (hunk 2 above)
Comment (Wilfred) [reply rust_src/src/process.rs]: You mean the C macros `SERIALCONN1_P` etc? It's used in 5 places, so it's a tradeoff between being explicit and being concise. I'm open to persuasion either way :)

File: rust_src/src/process.rs (hunk 3 above)
Comment (shaleh) [reply rust_src/src/process.rs]: It was a place holder. I meant to remove it.

**Comment by shaleh** [comment]
Keeping the name should make rollups easier.

I am predominantly a Python code. Explicit is always better than implicit. Names have value
-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

File: rust_src/src/process.rs (hunk 2 above)
Comment (brotzeit) [reply rust_src/src/process.rs]: Would it be a problem to compare `Lisp_Object`s here to avoid the casting ?

Comment (shaleh) [reply rust_src/src/process.rs]: My comment went out of band.

Keeping the name should make rollups easier. I am predominantly a Python code. Explicit is always better than implicit. Names have value

Comment (brotzeit) [reply rust_src/src/process.rs]: I think `Qserial` etc. is explicit enough for now, but the if statement is too verbose for what is done here.

**Comment by brotzeit** [comment]
Why did the mac test fail and the linux test not ?

```
4 | use remacs_sys::{EmacsInt, Lisp_Process, QCbuffer, Qcdr, Qclose, Qexit, Qlistp, Qnetwork, Qopen,

  |                                                          ^^^^^^ no `Qclose` in the root. Did you mean to use `Qclosed`?
```

File: src/process.c [hunk 4]
```diff
@@ -1061,52 +1061,6 @@ nil, indicating the current buffer's process.  */)
...
-  p = XPROCESS (process);
-  if (p->raw_status_new)
-    update_status (p);
-  status = p->status;
-  if (CONSP (status))
-    status = XCAR (status);
-  if (NETCONN1_P (p) || SERIALCONN1_P (p) || PIPECONN1_P (p))
-    {
-      if (EQ (status, Qexit))
-	status = Qclosed;
```
[the remark this answers is not in the document]
Comment (brotzeit) [reply src/process.c]: Yeah I know. But why did only the mac test reveal this ?

[the remark this answers is not in the document]
Comment (shaleh) [reply src/process.c]: I just tried this branch locally. It builds on this computer.

[the remark this answers is not in the document]
Comment (brotzeit) [reply src/process.c]: But you also use linux ? Take a look at the travis builds. This seems strange to me.

[the remark this answers is not in the document]
Comment (shaleh) [reply src/process.c]: Yes, I have built it this morning on both platforms.

[the remark this answers is not in the document]
Comment (shaleh) [reply src/process.c]: I saw that build. It is weird. Note that the clone even failed once. I wonder if there was a temporary weirdness in that virtual environment.

[the remark this answers is not in the document]
Comment (brotzeit) [reply src/process.c]: Why shouldn't be `Qclose` in the root ? Fortunate weirdness :)

[the remark this answers is not in the document]
Comment (shaleh) [reply src/process.c]: Basically globals.rs which is a generated file did not have Qclose in it. Why? Dunno.

[the remark this answers is not in the document]
Comment (brotzeit) [reply src/process.c]: Hidden debugger ;)

[the remark this answers is not in the document]
Comment (shaleh) [reply src/process.c]: :-)

**Comment by brotzeit** [comment]
I think this should be ok now.

**Comment by brotzeit** [comment]
Anything else ?

**Pull request merged**